### DISCUSSION TITLE
What features will not work when no fonts are available?

### DISCUSSION THREAD

**Discussion by seanpk** [mozman/ezdxf#988] [Q&A] [answered]
I'm am building an application that uses `ezdxf` in the browser (thanks to `pyodide`).

With the upgrade of `ezdxf` from `v1.0.3` to `v1.1.3` I see this error when I call functions that relate to text:
```
WARNING:ezdxf:no default font found: no fonts available, not even fallback fonts
```
Even with this warning, I am still able to export `MultiLeader`s to a DXF file and have other applications render the text.  

I am just starting on the text features and have not yet got to importing text from a DXF file, so I don't know whether that works or not. Since that is a feature I will need to implement, as well as many others, **does anyone know what features work/ought to work/won't work when `fonttools` can find no fonts?**

The answer to that question will be very helpful in assessing the value of figuring out how to make font files available to `fontools` in web assembly. Whether they be `.ttf` that I ship with the web app, or `.woff` that I pull from a CDN (which would mean making sure that the optional `.woff` support is available for `fonttools`), I don't think this is trivial work and would be happy to defer it or avoid it all together! :-)

Thanks

**Comment by mozman** [comment, marked as answer]
Hi!

`Ezdxf`  is designed to run in a local environment with access to the system fonts stored in the local file system (Windows, MacOS and Linux) and does not load fonts on the fly from the internet. Without access to the fonts all text-size calculations are done with a very inaccurate measurement font where each character has the same height and width.  **Everything works** but the result will look ugly and the `drawing` add-on will only render tofu ▯ for every character.

**Reply by seanpk** [reply]
thank you for the prompt answer
